sdk/storage/azure-storage-blob/azure/storage/blob/_serialize.py

```diff
@@ -176,50 +171,38 @@ def serialize_blob_tags_header(tags=None):
     return ''.join(components)
 
 
-def serialize_blob_tags(tags=None):
-    # type: (Optional[Dict[str, str]]) -> Union[BlobTags, None]
+def serialize_blob_tags(tags: Optional[Dict[str, str]] = None) -> Union[BlobTags, None]:
     tag_list = []
     if tags:
         tag_list = [BlobTag(key=k, value=v) for k, v in tags.items()]
     return BlobTags(blob_tag_set=tag_list)
 
 
-def serialize_query_format(formater):
+def serialize_query_format(formater: Union[str, DelimitedJsonDialect, "Dialect"]) -> Optional[QuerySerialization]:
     if formater == "ParquetDialect":
-        qq_format = QueryFormat(
-            type=QueryFormatType.PARQUET,
-            parquet_text_configuration=' '
-        )
+        qq_format = QueryFormat(type=QueryFormatType.PARQUET, parquet_text_configuration=None)
     elif isinstance(formater, DelimitedJsonDialect):
-        serialization_settings = JsonTextConfiguration(
-            record_separator=formater.delimiter
-        )
-        qq_format = QueryFormat(
-            type=QueryFormatType.json,
-            json_text_configuration=serialization_settings)
-    elif hasattr(formater, 'quotechar'):  # This supports a csv.Dialect as well
+        json_serialization_settings = JsonTextConfiguration(record_separator=formater.delimiter)
+        qq_format = QueryFormat(type=QueryFormatType.JSON, json_text_configuration=json_serialization_settings)
+    elif hasattr(formater, 'quotechar') and isinstance(formater, Dialect):  # This supports a csv.Dialect as well
         try:
-            headers = formater.has_header
+            headers = formater.has_header  # type: ignore
```

Comment:
azure\storage\blob\_serialize.py:197: error: Item "str" of "Union[str, Dialect]" has no attribute "has_header"  [union-attr]

I'm not sure how this is supposed to ever function, looking at documentation `has_header` is only a member of csv.Sniffer and not csv.Dialect: https://docs.python.org/3/library/csv.html#csv.Sniffer.has_header

Reply:
My guess would be that we actually support more than the type says.